File: ea/src/evolution_engine/mutators/Crossover.java
package evolution_engine.mutators;

import evolution_engine.evolution.Individual;

import java.util.List;

/**
 * Created by Itay Azaria
 * Date: 28/03/14
 * Time: 15:57
 */
public interface Crossover<I extends Individual> {


    public abstract List<I> cross(I father, I mother);

    public double getProbability();
}

File: evolution/src/main/java/bgu/cs/evolution_engine/mutators/Crossover.scala
package bgu.cs.evolution_engine.mutators

import bgu.cs.evolution_engine.evolution.Individual

/**
 * Created by Itay Azaria
 * Date: 28/03/14
 * Time: 15:57
 */
trait Crossover[I <: Individual] {
  def cross(father: I, mother: I): List[I]

  def getProbability: Double
}